test/sparse/sparsevector.jl

```diff
@@ -176,7 +176,23 @@ end
             I = rand(1:length(x), 20)
             r = x[I]
             @test isa(r, SparseVector{Float64,Int})
-            @test all(nonzeros(r) .!= 0.0)
+            @test all(!iszero, nonzeros(r))
+            @test Array(r) == Array(x)[I]
+        end
+
+        let x = convert(SparseVector{Float64,UInt32},sprandn(100,0.5))
```

Comment:
Same for the tests here, I think it would be nice with a comment what the test is for, and perhaps a reference to the PR/issue.